sdk/java-sdk/src/main/scala/kalix/javasdk/impl/DiscoveryImpl.scala

```diff
@@ -90,6 +82,27 @@ class DiscoveryImpl(
       in.proxyPort,
       in.protocolMajorVersion,
       in.protocolMinorVersion)
+
+    // possibly filtered or hidden env, passed along for substitution in descriptor options
+    val env: Map[String, String] = system.settings.config.getAnyRef("kalix.discovery.pass-along-env-allow") match {
+      case false => Map.empty
+      case true  => sys.env
+      case allowed: util.ArrayList[String @unchecked] =>
+        allowed.asScala.flatMap(name => sys.env.get(name).map(value => name -> value)).toMap
+    }
```

Comment:
so, if the property gets configured wrongly (for instance, passing a string directly) this throws and discovery breaks I guess? User would know that way, right? or should we help them with a specific message maybe?